src/cargo/core/compiler/mod.rs

```diff
@@ -33,14 +33,14 @@
 
 pub mod artifact;
 mod build_config;
-mod build_context;
+pub(crate) mod build_context;
```

Comment:
Not really a big deal. Just a bit worried making these modules public. Maintainer now need to gatekeep any misuse during the review.